src/services/refactors/helpers.ts

```diff
@@ -0,0 +1,25 @@
+/* @internal */
+namespace ts.refactor {
+    /**
+     * Returned by refactor funtions when some error message needs to be surfaced to users.
+     */
+    export interface RefactorErrorInfo {
+        error: string;
+    };
+
+    /**
+     * Checks if some refactor info has refactor error info.
+     */
+    export function isRefactorErrorInfo(info: unknown): info is RefactorErrorInfo {
+        return (info as RefactorErrorInfo).error !== undefined;
+    }
+
+    /**
+     * Checks if string "known" begins with string "requested".
+     * Used to match requested refactorKinds with a known refactorKind.
+     */
+    export function refactorKindBeginsWith(known: string, requested: string | undefined): boolean {

```

Comment:
Yep. I experimented with something fancier but it didn't seem to add anything except code complexity.

it would be sorta nice if we could provide string completions for the refactor kinds, but I couldn't think of anything besides just listing them all out in a big union in the `refactorKind` type.